test/inversify.test.ts

```diff
@@ -2808,4 +2808,89 @@ describe("InversifyJS", () => {
 
     });
 
+    describe("Injection of values with multiple tags", () => {
+        it("Should be able to resolve instance", () => {
+
+            interface Weapon {
+                use(): string;
+            }
+
+            @injectable()
+            class Katana implements Weapon {
+                public use() {
+                    return "katana!";
+                }
+            }
+
+            @injectable()
+            class Shuriken implements Weapon {
+                constructor(private readonly spikes: number) { }
+                public use() {
+                    return `${this.spikes} spikes shuriken!`;
+                }
+            }
+
+            const container = new Container();
+            const throwableConstraint = taggedConstraint('throwable')
+            const spikesConstraint = taggedConstraint('spikes')
+
+            const allConstraints = (...constraints: interfaces.ConstraintFunction[]) =>
+                (request: interfaces.Request) => constraints.every(constraint => constraint(request))
+
+            container.bind<Weapon>("Weapon").toConstantValue(new Shuriken(3))
+                .when(allConstraints(throwableConstraint(true), spikesConstraint(3)))
```

Comment:
I think there should be an additional whenTargetMultiTagged

Reply:
I though about creating that, I'll add it.

Reply:
Quick question about this.

When I use the `whenTargetMultiTagged` does the depdency should only be resolved when trying to resolve a dependency with all the tags or if it matches a single tag it is already enough to resolve that dependency? Basically my question here is if we should use `and` or `or` for the tags condition.

Reply:
It is a good point.
If there was to be just this change then a boolean would work.
Although we need to consider the parent and ancestor versions of these.

We could change below to accept multiple tags in a non breaking manner as you did for getTagged etc
These would be or - when having one of these tags
whenTargetTagged
whenParentTagged
whenAnyAncestorTagged
whenNoAncestorTagged

Then a multi tagged version for each of them - these being and.

@PodaruDragos 
@notaphplover 

Your thoughts gentlemen.

Reply:
I've implemented the `whenTargetMultiTagged` with the `or` behaviour on the definition but forcing the value to have all the tags present in the request.